src/class/emitter.helper.ts

```diff
@@ -0,0 +1,23 @@
+import * as forge from 'node-forge';
+
+export default class EmitterHelper {
+  public static getEmitterFromPfx(pfx: Buffer, passphrase: string): string | undefined {
+    let organizationalUnit;
+    try {
+      const p12Der = forge.util.decode64(pfx.toString('base64'));
+      const p12Asn1 = forge.asn1.fromDer(p12Der);
+      const p12 = forge.pkcs12.pkcs12FromAsn1(p12Asn1, false, passphrase);
+      const bags = p12.getBags({ bagType: forge.pki.oids.certBag });
+      const certBag = bags[forge.pki.oids.certBag];
+      if (!certBag) { return undefined; }
+      const { cert } = certBag[0];
+      if (!cert) { return undefined; }
+      organizationalUnit = cert.issuer.getField('OU').value;
+    } catch {}
```

Comment:
It should throw if it does not work

Reply:
it will return undefined, then throw

Reply:
What I mean is it should throw if there is one thing that is not expected:
- if there is no certBag
- if the certBag is empty
- if the OU field is empty

Think about how hard it is to debug a certificate error without proper error reporting. Make sure it throws as soon as possible, no need to wrap everything in a try/catch